app.js

```diff
@@ -1499,6 +1499,180 @@ app.delete("/room/:roomId", function (req, res) {
     });
 });
 
+
+//suggest meeting times given list of participants
+app.post("/api/meetingSuggestion", function (req, res) {
+
+    //currently only look 3 days ahead, and assume 7AM - 5PM workday. only search on weekdays
+
+    if(!req.body.participants) {
+      res.statusCode = 400;
+      console.log();
+      res.json({
+        "requestURL":  "/room",
+        "action": "post",
+        "status": 400,
+        "message": "Bad Request",
+        "timestamp": new Date()
+      });
+    }
+
+    else {
+
+      var participants = req.body.participants || null;
+
+      //using ADDDATE(CURDATE(), 4) just to limit the amount of meetings to search through later
+      var getMeetingSql = "SELECT DISTINCT(start_datetime), end_datetime FROM ebdb.meeting WHERE start_datetime >= CURDATE() AND end_datetime < ADDDATE(CURDATE(), 4) AND "
+                         + "calendar IN (SELECT primary_calendar FROM ebdb.user WHERE email IN (?)) ORDER BY end_datetime;";
+      var inserts = [participants];
+
+      pool.query(getMeetingSql, inserts, function(error, results, fields) {
+        if(error) {
+          res.statusCode = 500;
+          console.log(error);
+          res.json({
+            "requestURL":  "/meetingSuggestion",
+            "action": "post",
+            "status": 500,
+            "message": "Query failed for calendar",
+            "timestamp": new Date()
+          });
+        }
+        else {
+          
+          var meetings = results;
+
+          var parameters = {
+            meetings: meetings,
+            numDaysAhead: req.body.numDaysAhead,
+            startTime: req.body.startTime,
+            endTime: req.body.endTime
+          };
+
+          var timetable = createTimetable(parameters);
+
+          var countTimes = 0;
+          var suggestions = [];
+          //iterate through timetable and find first 5 suggestions
+          for(var time in timetable) {
+            if(countTimes == 5) {
+              break;
+            }
+            if(timetable[time] == 0) {
+              countTimes++;
+              suggestions.push(time);
+            }
+          }
+          //console.log(timetable);
+          //console.log(suggestions);
+          res.send([meetings, timetable, suggestions]);
+
+        }
+      });
+
+    }
+
+});
+
+function createTimetable(parameters) {
+
+    var meetings = parameters.meetings || null;
+    //var workdays = parameters.workdays || [1,2,3,4,5]; //list of workdays in the week (0-6)
```

Comment:
adding this would complicate things....

Reply:
Were you referring to the workdays parameter?  I think you could (isoweek())[https://momentjs.com/docs/#/get-set/iso-weekday/] from (use moment-js)[https://momentjs.com/] to determine the day of week and not add a meeting to the suggested meetings list if the startime is mapped to a saturday or sunday.